fastn-ds/src/lib.rs

```diff
@@ -124,8 +124,8 @@ impl DocumentStore {
         }
     }
 
-    pub async fn capture_event(
-        &self,
+    pub fn capture_event(
+        &mut self,
```

Comment:
Why do we need this? Can cause problem everywhere.

Reply:
We should use Dashmap instead of Hashmap if you are worried about collecting data. It allows modifications on read only references.